### PR TITLE
[ci] configure Bazel cache for GitHub actions

### PR DESCRIPTION
**Pull request by nbdd0121**
This should not be merged yet until we have configured bucket permission so that workload identity federation can write to the cache properly.

### CODE REVIEW

File: .github/actions/configure-bazel/action.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+runs:
+  using: composite
+  steps:
+    - name: Populate ~/.bazelrc
+      run: |
+        cp ci/.bazelrc ~/.bazelrc
+        # Inject the OS version into a parameter used in the action key computation to
+        # avoid collisions between different operating systems in the caches.
+        # See #14695 for more information.
+        echo "build --remote_default_exec_properties=OSVersion=\"$(lsb_release -ds)\"" >> ~/.bazelrc
```
Comment (jwnrt) [inline .github/actions/configure-bazel/action.yml:17]: Hmm, I wonder if this is still an issue. Can Bazel really not use the cache between glibc versions? Does it link things like `opentitantool` against the system glibc?

**Review by jwnrt — APPROVED** [review]
Looks good to me, thanks

File: .github/actions/configure-bazel/action.yml (hunk 1 above)
Comment (nbdd0121) [reply .github/actions/configure-bazel/action.yml:17]: A binary built against new glibc in general cannot be used with an old libc, so we cannot use binaries built on Ubuntu 22.04 on Ubuntu 20.04, for example.

Comment (jwnrt) [reply .github/actions/configure-bazel/action.yml:17]: I was hoping that Bazel's hermetic builds also included the libc, but it seems not

**Comment by jwnrt** [comment]
I've added the WIF pricipal to the bucket (I think), can I test this by pushing to a branch on the repo?

**Comment by jwnrt** [comment]
Good news is that it works, bad news is that some of the jobs are taking a lot longer:

* Verilated English Breakfast is very consistently 12m, but this run took ~16m.
* Lint (slow) is consistently 20m, but this run took over 30m just for the documentation build.

Any idea why it would be slowed down so much?

**Comment by jwnrt** [comment]
Ran again and it took much less time. The first run was either a transient error or it had to repopulate the cache which took a long time.

https://github.com/lowRISC/opentitan/actions/runs/11661952950

**Review by jwnrt — APPROVED** [review]
LGTM, but could you bring back the `- name: Prepare environment` lines? They're easier to read than `./.github/actions/prepare-environment` in the UI

**Pull request merged**